src/xenia/patcher/patch_db.cc

```diff
@@ -0,0 +1,164 @@
+/**
+ ******************************************************************************
+ * Xenia : Xbox 360 Emulator Research Project                                 *
+ ******************************************************************************
+ * Copyright 2022 Ben Vanik. All rights reserved.                             *
+ * Released under the BSD license - see LICENSE in the root for more details. *
+ ******************************************************************************
+ */
+#include <regex>
+
+#include "xenia/base/cvar.h"
+#include "xenia/base/filesystem.h"
+#include "xenia/base/logging.h"
+#include "xenia/memory.h"
+
+#include "xenia/patcher/patch_db.h"
+
+DEFINE_bool(apply_patches, true, "Enables patching functionality", "General");
+
+namespace xe {
+namespace patcher {
+
+PatchDB::PatchDB(const std::filesystem::path patches_root) {
+  patches_root_ = patches_root;
+  LoadPatches();
+}
+
+PatchDB::~PatchDB() {}
+
+void PatchDB::LoadPatches() {
+  if (!cvars::apply_patches) {
+    return;
+  }
+
+  const std::filesystem::path patches_directory = patches_root_ / "patches";
+  const std::vector<xe::filesystem::FileInfo>& patch_files =
+      filesystem::ListFiles(patches_directory);
+  const std::regex file_name_regex_match =
+      std::regex("^[A-Fa-f0-9]{8}.*\\.patch$");
+
+  for (const xe::filesystem::FileInfo& patch_file : patch_files) {
+    // Skip files that doesn't have only title_id as name and .patch as
```

Comment:
Not sure if this is right...
```suggestion
      std::regex("^[A-F0-9]{8}.*\\.patch\\.toml$");

  for (const xe::filesystem::FileInfo& patch_file : patch_files) {
    // Skip files that doesn't have only title_id as name and .patch.toml as
```